### PR TITLE
[Ingest Manager] Agent includes pgp file

### PR DESCRIPTION
**Pull request by michalpristas**
## What does this PR do?

This PR introduces baked in PGP file with a flag.
If `DEV=true` is specified during `mage build` PGP is not included and checks are omitted . 
Otherwise PGP is provided and passing check is required.

The solution works well with connected agent to internet, but locally baked in packages are a bit tricky as we need to find a way of including asc files into agent package so they can be checked.

## Why is it important?

More security running external binaries

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

### CODE REVIEW

**Comment by michalpristas** [comment]
related to this is including ASC files for beats into our package.
these are generated during release process so we need to figure out how. therefore this is still a draft with commented out ASC checks. HTTP downloader/verifyier works ok. problem is only with prepared packages

I contacted Chris to figure this out and come up with feasible strategy which would also help us with including endpoint probably

**Comment by blakerouse** [comment]
I think this looks good. By default PGP is used and required, only `DEV=true` at build time allows that to be removed.

**Comment by michalpristas** [comment]
not activating FS verifier just yet as for: https://github.com/elastic/infra/issues/21487
i think we can keep HTTP one turned on

**Comment by michalpristas** [comment]
*talking to infra about including asc files not merging before then*

**Comment by ph** [comment]
@michalpristas I think everything is in place in infra to move this forward?

**Comment by ph** [comment]
@michalpristas This should work with nightly snapshot too correct? When this get in, we need to communicate it with the other teams.

**Comment by michalpristas** [comment]
should work with snapshots yeah

File: x-pack/elastic-agent/magefile.go [hunk 1]
```diff
@@ -620,6 +625,23 @@ func buildVars() map[string]string {
...
 	vars["github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/release.snapshot"] = isSnapshot
 
+	fetchPgp := true
+	if isDevFlag, devFound := os.LookupEnv(devEnv); devFound {
+		if isDev, err := strconv.ParseBool(isDevFlag); err == nil && isDev {
+			vars["github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/release.allowEmptyPgp"] = "true"
+			fetchPgp = false
+		}
+	}
+	fmt.Println("fetching pgp", fetchPgp)
```
Comment (ruflin) [inline x-pack/elastic-agent/magefile.go:635]: leftover

Comment (michalpristas) [reply x-pack/elastic-agent/magefile.go:635]: this whole section will get removed

**Review by blakerouse — APPROVED** [review]
Looks good.

Really like the `DEV` env for development. Will actually be able to use that for the `install/uninstall` for self-upgrading, so in dev mode self-upgrading can be tested without it being installed in the correct paths. +1

**Comment by ph** [comment]
@michalpristas @blakerouse @ruflin we will need to document the `DEV` usage in the docs.

**Review by scunningham — APPROVED** [review]
Thanks. Question; does the PGP Key pull down during each release build; or we do we used a checked-in PGP Key unless somebody explicitly runs the code that pulls down the key and embeds it?

Pulling down the key on the fly during a build slightly concerns me as it allows an admittedly narrow window for attack. Subsequent unit tests should catch an injection, but we should validate that.

**Comment by michalpristas** [comment]
@scunningham nono key is prepacked in this PR and once we will need it updated we will update the file and run `mage update` for it to take effect

**Comment by michalpristas** [comment]
Faliures not related

**Pull request merged**